### ISSUE TITLE
Stop working after a few times

### ISSUE THREAD

**Issue by mickaphd** [aahung/Unshaky#110]
I use this app from a  long time now and I  just noticed (maybe from the last update) that after a few times Unshaky stop working. The app is still running but I still have the  keyboard problems and the debug  window do not display anything. I was able to make it work again after using Onyx but again, still not working after a few minutes. No idea if it's coming  from my  mac or the app.
PS: sorry  for the  double spaces but without Unshaky this butterfly keyboard is a nightmare...

**Comment by aahung** [comment]
Does an old version also have this issue? Can you try a older version?

**Comment by Zandor300** [comment]
Bug might have been introduced in version 0.5.4. Not later since no changes were made to the core logic of Unshaky since that version.

In 0.5.4, alternative keyboard layouts were added:
https://github.com/aahung/Unshaky/compare/v0.5.3...v0.5.4

**Comment by mickaphd** [comment]
I completely uninstall the 0.5.6 to reinstall the previous one and update again (after restart, cache cleaning etc.). It seems working now so I have no explanations. Maybe a bug from my mac.. You can close the issue and just wait for similar feedbacks maybe

**Comment by mickaphd** [comment]
Damn it's not working anymore...! I really  don't get it..

**Comment by aahung** [comment]
@Zandor300 can you confirm all versions since 0.5.4 do not work anymore?

@mickaphd so you are using the latest 0.5.6 and nothing shows up in the debug window? What OS version are u on now?

**Comment by mickaphd** [comment]
I use  the last Mojave version (10.14.6). 
I have the  lastest yes but even when I downgraded to the previous one at a certain  point it was not  working anymore. And yes when I feel  it's not working anymore like right now nothing appear in the debug window

**Comment by aahung** [comment]
@mickaphd Can you quickly try to set the delay for all (use the special input field below the table in Unshaky preference) to a large delay like 200 ms, to make sure it is not a delay configuration problem?

**Comment by Zandor300** [comment]
@aahung I personally didn’t experience/notice this bug. I was just looking up what might be the issue.

**Comment by aahung** [comment]
@Zandor300 thanks for looking out. I hope it is not a widespread issue

**Comment by mickaphd** [comment]
It  doesn't change anything. Do you know guys another  similar app/program?  I don't  wanna change for sure,  just  to  see if it's coming from my  computer and not Unshaky. Still thanks much for your help!

**Comment by aahung** [comment]
I found a similar tool named [debounce-mac](https://github.com/toothbrush/debounce-mac) which use the same API Unshaky does. Can you try to see it produce any output? 
Besides, can you do me a favor also test it when you don't use `sudo` to run that program (Unshaky usually don't acquire root privilege)?

**Comment by tuananh** [comment]
0.5.3 works for me. newer update do not.

**Comment by aahung** [comment]
@tuananh when it does not work, anything shows up in the debug window?

**Comment by tuananh** [comment]
its blank

**Comment by mickaphd** [comment]
Is it possible that the problem appears when the mac wakes up after sleeping? I mean I still don't find any logic explanations to explain why sometimes it works and sometimes not

**Comment by tuananh** [comment]
> Is it possible that the problem appears when the mac wakes up after sleeping? I mean I still don't find any logic explanations to explain why sometimes it works and sometimes not

for me, it doesn't work from the latest version. I can't find out why until I saw this issue. So i tried every version from latest down to 0.5.3 and that's when it starts working for me.

**Comment by mickaphd** [comment]
Can you try to update to the last version to see if it still works? Because in my case the most recent version still works sometimes (and just stop sometimes too..!), just to confirm if it's really a problem between 0.5.3 and 0.5.6

**Comment by aahung** [comment]
Interesting, @mickaphd I made a few changes about keyboard layout since 0.5.3, that might be the issue? @tuananh what keyboard layout do you use? Also did you use the keyboard layout feature in Unshaky configuration?

**Comment by aahung** [comment]
but the keyboard layout thing shouldn't affect the core function. One possibility I can think of is with the new keyboard layout feature, the delays might not be set properly, therefore, the debug window is empty (because it only shows events for keys with >0 delays). I think I should change this behavior to show EVERYTHING when debug window is open.

**Comment by tuananh** [comment]
> Interesting, @mickaphd I made a few changes about keyboard layout since 0.5.3, that might be the issue? @tuananh what keyboard layout do you use? Also did you use the keyboard layout feature in Unshaky configuration?

just the default us keyboard layout.

**Comment by mickaphd** [comment]
Just to let you know in case it can help to understand this issue, Unshaky still stop working for me sometimes after a few hours but I just noticed that if I lock my computer (lock screen, it's not needed to log out or shutdown), it works perfectly just after. I tried that from 3 days and it works. I have no explanation but maybe you will!

**Comment by clickysteve** [comment]
I have the same issue here. App doesn't appear to be working any more, even with very large delays. 0.5.4 version. OSX 10.13.6. Blank debug window.

**Comment by aahung** [comment]
#128 gives an possibility that Unshaky might conflict with other programs. I am not sure if it is the case for this issue though. 

@clickysteve when did it start to fail? Does upgrading to 0.6.0 resolve the problem?

**Comment by clickysteve** [comment]
> @clickysteve when did it start to fail? Does upgrading to 0.6.0 resolve the problem?

Just noticed it this past week. Upgrading to 0.6.0 hasn't resolved the issue. Seems to be a conflict with Karabiner.

**Comment by richardsimko** [comment]
I also have the same issue, it used to work fine but has now stopped working, can't really recall when. I don't use Karabiner and I don't have any other apps which mess with keys. Tried locking the screen but that didn't help. Let me know if I can do anything to help troubleshoot.

**Comment by richardsimko** [comment]
After an upgrade to Catalina it now works again!

**Comment by hazarek** [comment]
It just happened to me, when I set the millisecond, I realized it was ineffective.
When I restarted the app it worked properly. Today four more keys have begun to cause problems. and I broke the left shift, shift is working for now. `r` prints three times (thanks aggressive mode). I'm sure it's not related to this piece of dust. this is a hardware problem. sorry for my english.
Unshaky 0.6.0 Macos Catalina 10.15.1

**Comment by mariodian** [comment]
Stopped working for me too. I even set the delay to 999ms and it isn't blocking any double space whatsoever. 

I'm not entirely sure but I think it stopped working with the Catalina upgrade.

**Comment by eudinaesis** [comment]
I may be having the same issue. Mojave 10.14.6, MBAir 2018, English-International layout. I just installed Unshaky for the first time yesterday (0.6) and even when I set the delay very high, nothing happens -- blank debug window, no suppressed strokes.

**Comment by mickaphd** [comment]
Yes there is definitely a problem. For me when it stop working I just "Lock Screen" and just after unlocking it start working again. But it's really annoying. Try to see if it works for you

**Comment by aahung** [comment]
Sadly this has been a weird bug I can never reproduce 😢 . If you click the unshaky icon on the menu bar, it will try to restart all hooks. If that does not work (blank debug window when pressing configured keys), make sure you disable "ignore external keyboard" options. 

another workaround: remove Unshaky and download again to see if that helps (if it is caused by Unshaky cannot get accessibility permissions, reinstall Unshaky might ask you to grant permission again).

**Comment by mariodian** [comment]
Downgrading to 0.5.3 fixes the problem.

**Comment by ElectricWarr** [comment]
I also have this issue, or something very similar:

- Unshakey Version 0.6.0
- macOS 10.15.3

When it isn't working (which is most of the time), the debug window ONLY shows keypresses for buttonns:

- Right shift
- Right alt
- Right command
- Left command
- Left alt
- Left control
- fn
- Left shift
- Caps Lock

Here's an example I made by pressing every key on each row left-right top-bottom:

```
1586879446.188739 Key( 59| 57|      CapsLock|         0| 40) E(12)
1586879447.897737 Key( 59| 56|       Unknown|    131072| 40) E(12)
1586879448.025996 Key( 59| 56|       Unknown|         0| 40) E(12)
1586879448.666435 Key( 59| 60|       Unknown|    131072| 40) E(12)
1586879448.814658 Key( 59| 60|       Unknown|         0| 40) E(12)
1586879449.806379 Key( 59| 63|       Unknown|   8388608| 40) E(12)
1586879449.913821 Key( 59| 59|       Unknown|   8650752| 40) E(12)
1586879449.917630 Key( 59| 63|       Unknown|    262144| 40) E(12)
1586879449.977786 Key( 59| 58|       Unknown|    786432| 40) E(12)
1586879449.981562 Key( 59| 59|       Unknown|    524288| 40) E(12)
1586879450.030329 Key( 59| 55|       Unknown|   1572864| 40) E(12)
1586879450.038210 Key( 59| 58|       Unknown|   1048576| 40) E(12)
1586879450.093721 Key( 59| 55|       Unknown|         0| 40) E(12)
1586879450.318421 Key( 59| 54|       Unknown|   1048576| 40) E(12)
1586879450.374175 Key( 59| 61|       Unknown|   1572864| 40) E(12)
1586879450.381945 Key( 59| 54|       Unknown|    524288| 40) E(12)
1586879450.429957 Key( 59| 61|       Unknown|         0| 40) E(12)
```

**Comment by aahung** [comment]
Thanks for the information. [this long-lasting issue makes me scratch my head so much!]

**Comment by ElectricWarr** [comment]
Is it normal for all of these keys to show up as "Unknown"? Can you think of any reason only these modifier keys are picked up, and nothing else, or rather is there anything you are aware of that groups these keys?

[Attached a config export](https://github.com/aahung/Unshaky/files/4481732/unshakey_config.txt) in case that answers any questions - this has stayed constant between the "working" and "not working" states.

**Comment by Bluekezhou** [comment]
您的邮件已收到，谢谢！
